src/sqlitestorage.cpp

```diff
@@ -1033,7 +541,16 @@ bool SqliteStorage::modifiedIncidences(Incidence::List *list, const QDateTime &a
 bool SqliteStorage::deletedIncidences(Incidence::List *list, const QDateTime &after,
                                       const QString &notebookUid)
 {
-    if (d->mDatabase && list) {
+    if (!after.isValid()) {
+        return d->mBackend.deletedIncidences(list, notebookUid);
```

Comment:
The SqliteStorage has methods for fetching deleted, modified and inserted events after a given time. The newer api seems only having this one for getting all the deleted incidences. 

Those could be still useful features to have? Also it could be nice if having implementation on the lower level would mean we could remove all sqlite api and query references on this file.

Reply:
> Those could be still useful features to have?

As far as I know, the `after` cases were used in the sync plugins to get the deleted (or inserted or modified) incidences since the last successful sync. But I think they are bad design (for such cases). Let me explain:
- sync the database with a distant server at time T0,
- delete (or insert, or modify) an incidence at T > T0,
- let the system sync again at T1 > T > T0 (usual case), but for any reason, the sync didn't succeed for all modifications to be sent to the server (you pass in a tunnel at the wrong moment). But the sync is considered as successful with minor errors. The sync date is thus increased to T1.
- in the next sync, the deleted incidence won't be reproposed.

There are several ways to fix this:
- don't increase the sync time when there are minor errors. But this is creating other issues, like inserted incidences that succeeded will be represented, except if you fiddle with their created dates in the database…
- increase the sync time for minor errors and fiddle with the deletion (or creation or modification) date for the failed ones so they will be represented at next sync…
- don't use the `after` argument in the listing routines.

In my opinion, only the last solution makes sense and avoid fiddling with the DB internals. In addition, from a user point of view, I don't want to get in sync with the changes since the last successful sync. I just want to be in sync, that's all. Which means that I want the sync plugin to list all deleted incidences in the device and propagate all the changes upstream. The same for added and modified incidences.

I've modified a long time ago the CalDAV plugin to use `deletedIncidences()` without `after` argument and to also use `allIncidences()` without it either. You can get all added incidences by looking for all incidences if they have an `etag` set… I know that the Google plugin is still using these `after` arguments. I still need to migrate it.

All in all, that's why I didn't move these `after` arguments into the new backend API. Without this argument, the `modifiedIncidences()` have no sense. So there is no trace of it in the new backend API. If you look where it is used (even in the Google plugin), there is a call to `allIncidences()` anyway. From these incidences, one can easily mimick the behaviour of `modifiedIncidences()` without needing to another DB call and the creation of another Incidence::List. One can just do the `WHERE` clause by hand with the `lastModified()` method of incidences.

> Also it could be nice if having implementation on the lower level would mean we could remove all sqlite api and query references on this file.

Indeed, all SQLite stuff should disappear from this level and be kept inside SingleSqliteBackend class. What remains here is  to support what I would call "deprecated" functions that I don't want to have support for in the backend. If you look at new classes in #61, there is no more SQL at this level.

Reply:
> don't use the after argument in the listing routines.

So in other words: figure out the modifications and not synced deletions separately? But then again those wouldn't be mutually exclusive. One could use the 'after' parameter as optimization for guaranteed known proper sync point and compare a subset of all the calendar data?

Reply:
As discussed on IRC, I'll add routines to support specific listing, as used before in the `dletedIncidences()`, `insertedIncidences()` and `modifiedIncidences()` functions.